src/TraceEvent/Stacks/Linux/LinuxPerfScriptEventParser.cs

```diff
@@ -432,36 +426,68 @@ private IEnumerable<LinuxEvent> NextEvent(Regex regex, FastStream source)
 
                     Frame threadTimeFrame = null;
 
-                    // For the sake of immutability, I have to do a similar if-statement twice. I'm trying to figure out a better way
-                    //   but for now this will do.
-                    ScheduleSwitch schedSwitch = null;
-                    if (eventKind == EventKind.Scheduler)
-                    {
-                        source.RestoreToMark(markedPosition);
-                        schedSwitch = ReadScheduleSwitch(source);
-                        source.SkipUpTo('\n');
-                    }
+                    // Event Properties
+                    // I mark a position here because I need to check what type of event this is without screwing up the stream
+                    var markedPosition = source.MarkPosition();
+                    source.ReadAsciiStringUpTo('\n', sb);
+                    string eventDetails = sb.ToString().Trim();
+                    sb.Clear();
 
-                    ThreadExit exit = null;
-                    if(eventKind == EventKind.ThreadExit)
+                    // Go back to the event so we can process it
+                    source.RestoreToMark(markedPosition);
+                    if (eventDetails.Length >= SchedulerEvent.Name.Length && eventDetails.Substring(0, SchedulerEvent.Name.Length) == SchedulerEvent.Name)
                     {
-                        source.RestoreToMark(markedPosition);
-                        exit = ReadExit(source);
-                        source.SkipUpTo('\n');
+                        ScheduleSwitch schedSwitch = ReadScheduleSwitch(source);
+                        IEnumerable<Frame> frames = ReadFramesForSample(processCommand, pid, tid, threadTimeFrame, source);
```

Comment:
No need to change in this PR - I think we should consider if there is a nice pattern we can employ here to ensure that the call to read the stack happens automatically for all event types.